Fix bug in list_hp_retire function

kevinshieh0225 opened the pull request:
In hp_list project's function : _list_hp_retire_, it attempt to reorganize retire list while the object is deleting.
However, as I see, the origin code doesn't adjust the index simultaneously, which will skip track of the object
origin from _&rl->list[iret + 1]_. I'm not sure whether the behavior was originally intended or not.

jserv commented:
@linD026, Please help review this proposed change.

jserv reviewed (changes requested):
Read https://cbea.ms/git-commit/ carefully. Then, write down informative commit message(s).

jserv commented:
You should use `git rebase` rather than `git merge`. Then, use `git rebase -i` to squash the commits and force-push.

kevinshieh0225 committed:
Fix bug in list_hp_retire function

In hp_list project's function : list_hp_retire, it attempts to reorganize
retire list while the object is deleting. However, as I see, the origin
code doesn't adjust the index simultaneously, which will skip track of
the object origin from &rl->list[iret + 1].

Change the behavior as:
 - If reorganize happens, keep the same index for next iteration.
 - Else, index++.

linD026 commented:
In my opinion, the original code might be a bug.
Since the delete is to move the objects after the `iret` forward one index, it will skip the next object when we increase the `iret` after every time deleted.

kdnvt commented:
I think the change is correct since the original code skips the next element if a deletion happens.

jserv commented:
Thank @kevinshieh0225 for contributing. The git commit message was amended upon the review of @linD026 and @kdnvt .
